receiver/vcenterreceiver/internal/metadata/generated_metrics_v2.go

```diff
@@ -465,7 +465,7 @@ func (m *metricVcenterClusterHostCount) init() {
 	m.data.Sum().DataPoints().EnsureCapacity(m.capacity)
 }
 
-func (m *metricVcenterClusterHostCount) recordDataPoint(start pcommon.Timestamp, ts pcommon.Timestamp, val int64, hostEffectiveAttributeValue string) {
+func (m *metricVcenterClusterHostCount) recordDataPoint(start pcommon.Timestamp, ts pcommon.Timestamp, val int64, hostEffectiveAttributeValue bool) {
```

Comment:
This change is happening due to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/vcenterreceiver/metadata.yaml#L39

Not sure if we should change `vcenterreceiver` in this PR itself or change `metadata.yaml` for `vcenterreceiver` to make it "backward-compat" and then fix it correctly.

Reply:
This is a good opportunity to test the changes. Looks like it doesn't work with `enum` field. let's remove these fields https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/vcenterreceiver/metadata.yaml#L40-L42 and handle enum + type use case separately

Reply:
> @dmitryax I think enum+type is working. for eg: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/vcenterreceiver/metadata.yaml#L32-L35. I am not understanding what you are suggesting with handle enum+type use case separately?

it's working only for string type, not for other types

Reply:
> Another question: If we fix vcenterreceiver to use bool type instead of enum; I think I need to update CHANGELOG.md as well and indicate this will be a breaking change for users?

This is right. Given that it's really a breaking change I would recommend doing it separately to keep clean git history. In that case we will need to remove this line https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/vcenterreceiver/metadata.yaml#L39 instead

Reply:
Makes sense to do a separate PR to keep git history clean.